src/spatial/spatial_massaction_jump.jl

```diff
@@ -0,0 +1,39 @@
+struct SpatialMassActionJump{T,S,U,V} <: AbstractMassActionJump
+    scaled_rates::T
+    reactant_stoch::S
+    net_stoch::U
+    param_mapper::V
+end
+
+# rx_coefficients[i,j] is the coefficient of reaction i at site j by which the reaction rate will be multiplied
+function SpatialMassActionJump(ma_jumps::MassActionJump{T,S,U,V}, rx_coefficients::Matrix) where {T,S,U,V}
```

Comment:
I'm confused about the use case for this constructor. If the user gives an existing `MassActionJump`, the `SpatialMassActionJump` shouldn't have any spatial matrix but just copy or alias the rate vector from the `MassActionJump`. (In fact, we should have a `nocopy=false` kwarg to allow control on whether these are copied over.)

Then, we should also have a separate `SpatialMassActionJump` constructor, analogous to the `MassActionJump` constructor, which has similar keywords and takes the various fields directly (along with scaling the rates). (Though we can ignore the parameter mapping stuff for now.)